test/ecto/query/builder/cte_test.exs

```diff
@@ -12,16 +12,24 @@ defmodule Ecto.Query.Builder.CTETest do
   test "fragment CTE" do
     query = "products" |> with_cte("categories", as: fragment("SELECT * FROM categories"))
 
-    assert [{"categories", %Ecto.Query.QueryExpr{expr: expr}}] = query.with_ctes.queries
+    assert [{"categories", false, %Ecto.Query.QueryExpr{expr: expr}}] = query.with_ctes.queries
     assert {:fragment, [], [raw: "SELECT * FROM categories"]} = expr
     refute query.with_ctes.recursive
   end
 
+  test "materialized CTE" do
```

Comment:
we should add a test for materialized = false as well once that change has been made